### PR TITLE
Feat/dao launcher treasury step

### PR DESCRIPTION
**Pull request by pnagarjun16**
(https://www.notion.so/rational-nomads/DAO-Launcher-Treasury-Step-51d9c67d5c24418aa392aba5965b3f7b)

-----

## 1. Pull request checklist

- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

## 2. PR type
    - Feature

## 3. What is the old behaviour?
Previously the treasury step for createDAO was empty.

## 4. What is the new behaviour?
Now it has a functioning treasury step where user can add gnosis safe that looks like the design.

### CODE REVIEW

**Comment by ashmidgley** [comment]
Great work on the changes @pnagarjun16! 🚀

Unfortunately it looks like we have a casing issue with the `Steps` folder where it's creating a new copy of the files in `steps`. Could you please run `git mv steps tempname && git mv tempname Steps` locally and push the changes? (See here for reference: https://stackoverflow.com/questions/11183788/in-a-git-repository-how-to-properly-rename-a-directory).

File: src/features/create-dao/Steps/TreasuryForm.tsx [hunk 1]
```diff
@@ -1,24 +1,56 @@
...
 import { Wizard } from '@zero-tech/zui/components';
 import { CreateDAOFormContext } from '../';
 
 import styles from './TreasuryForm.module.scss';
+import { WrappedInput } from '../../ui/WrappedInput/WrappedInput';
+import classNames from 'classnames/bind';
 
-interface TreasuryFormProps {
+const validationSchema = Yup.object().shape({
+	gnosisSafe: Yup.string().required('Gnosis Safe is required'),
```
Comment (ashmidgley) [inline src/features/create-dao/Steps/TreasuryForm.tsx]: Let's update the validation message here to 'The gnosis safe field is required.'.

File: src/features/create-dao/Steps/TreasuryForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+		fireEvent.blur(screen.getByPlaceholderText(/Gnosis Safe/i));
+		fireEvent.click(
+			screen.getByRole('button', {
+				name: 'Next',
+			}),
+		);
+
+		await waitFor(() =>
+			expect(screen.getByText('Gnosis Safe is required')).not.toBe(
```
Comment (ashmidgley) [inline src/features/create-dao/Steps/TreasuryForm.test.tsx]: Will also need to update this string to 'The gnosis safe field is required.'.

File: src/features/create-dao/Steps/TreasuryForm.tsx [hunk 3]
```diff
@@ -1,24 +1,56 @@
...
+			validationSchema={validationSchema}
+		>
+			{({ values, errors, touched, setFieldValue, submitForm }) => (
 				<Form>
-					<p>Enter treasury content here...</p>
+					<div>
+						<WrappedInput
+							label="What Ethereum address do you want all of the DAO’s funds to be held in?"
+							value={values.gnosisSafe}
+							placeholder="Gnosis Safe"
```
Comment (ashmidgley) [inline src/features/create-dao/Steps/TreasuryForm.tsx]: Not in line with the designs but for consistency with the other forms let's change this to 'Enter gnosis safe address...'.

File: src/features/create-dao/Steps/TreasuryForm.test.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+				<CreateDAOFormContext.Provider value={DEFAULT_PROVIDER_VALUES}>
+					<TreasuryForm {...DEFAULT_PROPS} />
+				</CreateDAOFormContext.Provider>
+			</ZUIProvider>
+		);
+
+		const user = userEvent.setup();
+
+		await user.type(
+			screen.getByPlaceholderText(/Gnosis Safe/i),
```
Comment (ashmidgley) [inline src/features/create-dao/Steps/TreasuryForm.test.tsx]: Will also need to update this to /Enter gnosis safe address.../i

File: src/features/create-dao/Steps/TreasuryForm.tsx [hunk 5]
```diff
@@ -1,24 +1,56 @@
 import { FC, useContext } from 'react';
 
+import * as Yup from 'yup';
 import { Form, Formik } from 'formik';
 
 import { Wizard } from '@zero-tech/zui/components';
 import { CreateDAOFormContext } from '../';
 
 import styles from './TreasuryForm.module.scss';
+import { WrappedInput } from '../../ui/WrappedInput/WrappedInput';
```
Comment (ashmidgley) [inline src/features/create-dao/Steps/TreasuryForm.tsx]: nit: let's reorder these imports to match zer0's preferred pattern: https://softwaresurgery.slack.com/archives/CQKH46ADB/p1666060086775939.

**Comment by ashmidgley** [comment]
@pnagarjun16 great work mate! Looking good. Just a couple small tweaks to address. Once you've done those please `fetch` the latest from `feat/dao-launcher-form-wireup` and rebase/squash into a single commit :)

File: src/features/create-dao/Steps/TreasuryForm.tsx (hunk 1 above)
Comment (pnagarjun16) [reply src/features/create-dao/Steps/TreasuryForm.tsx]: Updated

File: src/features/create-dao/Steps/TreasuryForm.test.tsx (hunk 2 above)
Comment (pnagarjun16) [reply src/features/create-dao/Steps/TreasuryForm.test.tsx]: Updated

File: src/features/create-dao/Steps/TreasuryForm.tsx (hunk 3 above)
Comment (pnagarjun16) [reply src/features/create-dao/Steps/TreasuryForm.tsx]: Updated

File: src/features/create-dao/Steps/TreasuryForm.test.tsx (hunk 4 above)
Comment (pnagarjun16) [reply src/features/create-dao/Steps/TreasuryForm.test.tsx]: updated

File: src/features/create-dao/Steps/TreasuryForm.tsx (hunk 5 above)
Comment (pnagarjun16) [reply src/features/create-dao/Steps/TreasuryForm.tsx]: Changed the order

**Pull request closed without merging**